tests/test_repro.py

```diff
@@ -122,7 +123,7 @@ def test(self):
 
             yaml.safe_dump(stage_dump, fd, default_flow_style=False)
 
-        with self.assertRaises(StagePathAsOutputError):
+        with self.assertRaises(OverlappingOutputPathsError):
```

Comment:
But this test is testing `StagePathAsOutputError`, not OverlappingOutputPathsError. Please don't change tests like this, you are breaking it. What we should do here is adjust dvc run commands, to have StagePathAsOutputError but not OverlappingOutputPathsError .

Reply:
Adjusted so that errors stay as they were